### PR TITLE
Add the daemon config UI

### LINKED ISSUE #830: Add GUI support for managing fapolicyd.config
Text editor similar to rules for editing and validating fapolicyd configuration.

### PR DESCRIPTION
**Pull request by egbicker**
Adds the user interface components for fapolicyd configuration editing and deployment.

Closes #830

### CODE REVIEW

**Comment by egbicker** [comment]
@jw3 I'm trying to implement the `ConfigChangeset` but even after rebuilding the rust layer I'm still getting `AttributeError: module 'fapolicy_analyzer' has no attribute 'ConfigChangeset'`

**Comment by jw3** [comment]
> @jw3 I'm trying to implement the `ConfigChangeset` but even after rebuilding the rust layer I'm still getting `AttributeError: module 'fapolicy_analyzer' has no attribute 'ConfigChangeset'`

I missed a registration call in #908.  #910 adds it and adds an example config changest usage.

**Comment by jw3** [comment]
Deployment nearly working, needs #911

**Comment by jw3** [comment]
@egbicker do you have tests coming for the new stuff?

**Comment by egbicker** [comment]
@jw3 Just got them pushed

File: fapolicy_analyzer/ui/changeset_wrapper.py [hunk 1]
```diff
@@ -42,13 +42,38 @@ def deserialize(cls, data: Union[Dict[str, str], str]) -> "Changeset":
...
         elif isinstance(data, str):
-            rcs = RuleChangeset()
-            rcs.parse(data)
-            return rcs
+            try:
+                rcs = RuleChangeset()
+                rcs.parse(data)
+                return rcs
 
+                try:
```
Comment (jw3) [inline fapolicy_analyzer/ui/changeset_wrapper.py:50]: Whats going on here?

**Review by jw3 — COMMENTED** [review]
This is looking pretty good. A few things

- [ ] Noted the nesting exception handling above for clarification
- [ ] I still saw the unsaved rule notification, but that looks like a problem with the rules view state management
- [ ] Can a summary of config changes be added to this view?  What bindings do we need added for this?
image

**Comment by jw3** [comment]
Added #912 describing some related concerns.  Doesnt have to be fixed for this PR, but could be if simple enough.

Another thought with this is that the unsaved rule edit check would apply to config as well.

File: fapolicy_analyzer/ui/changeset_wrapper.py (hunk 1 above)
Comment (egbicker) [reply fapolicy_analyzer/ui/changeset_wrapper.py:50]: It's trying to load the changeset data and now we have 2 `Changeset` that return as string and the `RuleChangeset` has additional exceptions in the rust layer (I think its something about having 0 rules)

**Comment by egbicker** [comment]
@jw3 I believe we'd need a binding like `rules_difference`

File: fapolicy_analyzer/ui/changeset_wrapper.py (hunk 1 above)
Comment (egbicker) [reply fapolicy_analyzer/ui/changeset_wrapper.py:50]: It was but not like that. Good catch.

**Comment by jw3** [comment]
#913 is on the way

**Pull request merged**